models/aggregate.py

```diff
@@ -0,0 +1,75 @@
+"""Aggregate data from S3 into ML ready format. WIP.
+"""
+import pandas as pd
+import pygeohash
+from pyarrow.parquet import ParquetFile
+from pyproj import Geod
+
+
+files = ['Mar10.parquet',
+         'Mar11.parquet',
+         'Mar12.parquet',
+         'Mar13.parquet',
+         'Mar14.parquet',
+         'Mar15.parquet',
+         'Mar16.parquet',
+         'Mar17.parquet',
+         ]
+
+for i, filename in enumerate(files):
+    print(f'\n{filename}')
+
+    datestring = filename.split('.')[0]
+    pf = ParquetFile('../data/' + filename)
+
+    df_case_grid = pd.DataFrame()
+
+    for j in range(pf.num_row_groups):
+        print(f'  processing group {j + 1}/{pf.num_row_groups}...')
+        # load one group at a time (its a big file)
+        df_raw = pf.read_row_group(j).to_pandas()
+
+        # calculate lat and lon
+        unique_geohash = df_raw[['geo_hash']].drop_duplicates()
+        lat_lon = pd.DataFrame(
+            unique_geohash.geo_hash.apply(pygeohash.decode).to_list(),
+            columns=['lat', 'lon'])
+        lat_lon['geo_hash'] = unique_geohash.values
+
+        # get reference points ONCE and use same ones every time
+        if i == 0:
+            lon_ref, lat_ref = lat_lon.lon.mean(), lat_lon.lat.mean()
```

Comment:
These are the "reference" points that will define the (0, 0) in this grid. Not sure if this the best way to do this, but I define it once and then use that same point for measuring distances in km to all cases.